### PR TITLE
@l2succes: [FEATURE Authentication] add analytics to mobile wizard steps

### PR DESCRIPTION
**Pull request by xtina-starr**
This addresses https://artsyproduct.atlassian.net/browse/GROW-684 and adds analytic events for successfully clicking the next button on the email and password steps of the mobile signup flow.

### CODE REVIEW

File: src/Components/Authentication/FormSwitcher.tsx [hunk 1]
```diff
@@ -165,6 +166,7 @@ export class FormSwitcher extends React.Component<FormSwitcherProps, State> {
 
     return (
       <Form
+        contextModule={this.props.options.contextModule}
```
Comment (xtina-starr) [inline src/Components/Authentication/FormSwitcher.tsx]: I actually think I need to pass the `intent` versus the `contextModule` per the spreadsheet in the ticket.

File: src/Components/Authentication/Mobile/SignUpForm.tsx [hunk 2]
```diff
@@ -18,17 +20,36 @@ import { ProgressIndicator } from "Components/ProgressIndicator"
...
+  1: Schema.ActionName.PasswordNextButton,
+}
+
+@track({ context_module: Schema.Context.Header })
+export class MobileSignUpForm extends Component<Props, MobileSignUpFormState> {
   state = {
     isSocialSignUp: false,
   }
 
+  @track((props, state, args) => ({
```
Comment (xtina-starr) [inline src/Components/Authentication/Mobile/SignUpForm.tsx]: I pass the `currentStepIndex` to the `trackNextClick` and access it via the args here.

File: src/Components/Authentication/Mobile/SignUpForm.tsx [hunk 3]
```diff
@@ -18,17 +20,36 @@ import { ProgressIndicator } from "Components/ProgressIndicator"
...
+    action_type: Schema.ActionType.Click,
+    action_name: currentStepActionName[args[0]],
```
Comment (l2succes) [inline src/Components/Authentication/Mobile/SignUpForm.tsx]: Nice use of `args` 👍

File: src/Components/Authentication/FormSwitcher.tsx [hunk 4]
```diff
@@ -18,6 +18,7 @@ import {
 } from "./Types"
 
 export interface FormSwitcherProps {
+  contextModule?: string
```
Comment (l2succes) [inline src/Components/Authentication/FormSwitcher.tsx]: Shouldn't `contextModule` come from `ModalOptions`?

File: src/Components/Authentication/FormSwitcher.tsx (hunk 1 above)
Comment (l2succes) [inline src/Components/Authentication/FormSwitcher.tsx]: See above ^

File: src/Components/Authentication/FormSwitcher.tsx (hunk 4 above)
Comment (xtina-starr) [reply src/Components/Authentication/FormSwitcher.tsx]: Yeah this isn't actually needed. I'm cleaning this up a bit.

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit dc7c43bb6967bbde7576d617d5e7c146661fd933

https://deploy-preview-1211--artsy-reaction.netlify.com

**Pull request merged**